components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/ActionExecutorServiceImpl.java

```diff
@@ -0,0 +1,340 @@
+import org.wso2.carbon.identity.action.execution.exception.ActionExecutionException;
+import org.wso2.carbon.identity.action.execution.internal.ActionExecutionServiceComponentHolder;
+import org.wso2.carbon.identity.action.execution.model.ActionExecutionRequest;
+import org.wso2.carbon.identity.action.execution.model.ActionInvocationErrorResponse;
+import org.wso2.carbon.identity.action.execution.model.ActionInvocationResponse;
+import org.wso2.carbon.identity.action.execution.model.ActionInvocationSuccessResponse;
+import org.wso2.carbon.identity.action.execution.model.ActionType;
+import org.wso2.carbon.identity.action.execution.model.AllowedOperation;
+import org.wso2.carbon.identity.action.execution.model.PerformableOperation;
+import org.wso2.carbon.identity.action.execution.util.APIClient;
+import org.wso2.carbon.identity.action.execution.util.AuthMethods;
+import org.wso2.carbon.identity.action.execution.util.OperationComparator;
+import org.wso2.carbon.identity.action.management.exception.ActionMgtException;
+import org.wso2.carbon.identity.action.management.model.Action;
+import org.wso2.carbon.identity.action.management.model.AuthProperty;
+import org.wso2.carbon.identity.action.management.model.AuthType;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * ActionInvocationService.
+ */
+
+public class ActionExecutorServiceImpl implements ActionExecutorService {
+
+    private static final Log log = LogFactory.getLog(ActionExecutorServiceImpl.class);
+
+    private static final ActionExecutorServiceImpl instance = new ActionExecutorServiceImpl();
+    private final APIClient apiClient;
+
+    private ActionExecutorServiceImpl() {
+
+        apiClient = new APIClient();
+    }
+
+    public static ActionExecutorServiceImpl getInstance() {
+
+        return instance;
+    }
+
+    public void execute(ActionType actionType, Map<String, Object> eventContext, String tenantDomain)
+            throws ActionExecutionException {
+
+        ActionExecutionRequest actionRequest =
+                ActionExecutionRequestBuilderFactory.buildActionExecutionRequest(actionType, eventContext);
+
+        List<Action> actions;
+        try {
+            actions = ActionExecutionServiceComponentHolder.getInstance().getActionManagementService()
+                    .getActionsByActionType(actionType.name(), tenantDomain);
+        } catch (ActionMgtException e) {
+            //todo: throw and block ?
+            log.warn("Skip executing actions for action type: " + actionType.name() +
+                    ". Error occurred while retrieving actions.");
+            if (log.isDebugEnabled()) {
+                log.debug("Error: " + e.getMessage(), e);
+            }
+
+            return;
+        }
+
+        if (actions.isEmpty()) {
+            log.info("No actions found for action type: " + actionType.name());
```

Comment:
Is this necessary to be an info log ?

Reply:
I also felt it's not needed. Removed and fixed